auth-source-pass.el

```diff
@@ -197,7 +196,15 @@ CONTENTS is the contents of a password-store formatted file."
 
 Disambiguate between user provided inside HOST (e.g., user@server.com) and
 inside USER by giving priority to USER.  Same for PORT."
-  (apply #'auth-source-pass--find-match-unambiguous (auth-source-pass--disambiguate host user port)))
+  (seq-some (lambda (a-host)
```

Comment:
I thought that passing multiple hosts to `auth-source` means that multiple auth-source entries can be returned. In your implementation, only one of them is returned. What is the purpose of the `:max` token in `auth-source-search` then? I find auth-source a bit confusing on this matter.

Reply:
I think `:max` should be handled independently from this multiple hosts.

For example, `auth-source-netrc` returns multiple results even for single host. (Actually, it returns inconsistent results as below on Emacs 26.2)

~/.authinfo
```
machine foo login user1 password pass1 port 22
machine foo login user2 password pass2 port imaps
machine bar login user3 password pass3 port 22
machine bar login user4 password pass4 port imaps
```

```emacs-lisp
(length (auth-source-search :host "foo")) ; => 1
(length (auth-source-search :host "foo" :max 2)) ; => 1 Is this bug?
(length (auth-source-search :host "bar" :max 2)) ; => 2
(length (auth-source-search :host '("foo") :max 2)) ; => 2
(length (auth-source-search :host '("foo" "bar") :max 4)) ; => 4
(length (auth-source-search :port "imaps" :max 2)) ; => 2
```

Reply:
Something (cache?) was wrong in my previous comment.

Restarted Emacs and `(length (auth-source-search :host "foo" :max 2))` is 2 as expected.

Reply:
> Restarted Emacs and (length (auth-source-search :host "foo" :max 2)) is 2 as expected.

As I understand it, your PR doesn't make auth-source-pass capable or returning more than one entry: it will always return at most one entry (because of [this](https://github.com/DamienCassou/auth-password-store/pull/99/files#diff-910e1cbfdd0014ca385796c7830a50d2R76) and [that](https://github.com/DamienCassou/auth-password-store/pull/99/files#diff-910e1cbfdd0014ca385796c7830a50d2R202)).

This is in contrast with the authinfo example you give above:

> `(length (auth-source-search :host '("foo" "bar") :max 4))` ; => 4

as, in this case, all entries matching either "foo" *or* "bar" are returned.

Do I understand correctly? Is that good enough?

Reply:
> As I understand it, your PR doesn't make auth-source-pass capable or returning more than one entry: it will always return at most one entry (because of this and that).

Right.

> This is in contrast with the authinfo example you give above:

The example is what `auth-source-netrc` does. My point here is that `auth-source-netrc` can return multiple entries even if single host is specified in the search spec. So I think it is better to take up `:max` handling separately from this PR.

Reply:
agree